qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1134,36 +1135,69 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 	 * @param projectUri  the project Uri.
 	 * @param diagnostics the diagnostic list.
 	 */
-	private void validateCaseSectionParameters(Section caseSection, Section whenSection, ResolvedJavaTypeInfo whenType,
-			String projectUri, List<Diagnostic> diagnostics) {
+	private void validateCaseSectionParameters(CaseSection caseSection, Section whenSection,
+			ResolvedJavaTypeInfo whenType, String projectUri, List<Diagnostic> diagnostics) {
 		String expression = whenSection.getExpressionContent();
 		List<Parameter> parameters = caseSection.getParameters();
 		String whenTypeName = whenType.getSignature();
+		Parameter firstParameter = parameters.get(0);
 		// Iterate through children and ensure the type of the #case or #is parts are
 		// the same as the parent section
 		for (int i = 0; i < parameters.size(); i++) {
 			Parameter parameter = parameters.get(i);
 			String parameterName = parameter.getName();
-			if (i == 0 && parameters.size() > 1) {
-				// The first is allowed as an operator so it is skipped for, second must be
-				// checked
-				// Ex: {#is in BROKEN}
+			Range range = QutePositionUtility.createRange(parameter);

```

Comment:
Don't create a range here since you could not use it. I suggest that you create the range only if you need it.